scripts/travis.sh

```diff
@@ -7,15 +7,21 @@ set -ev
 # set defaults if not already set
 PRINT_SUMMARY=${PRINT_SUMMARY:-true}
 RUN_ORDER=${RUN_ORDER:-filesystem}
+
 HEADLESS=${HEADLESS:-false}
 SKIPINTERMITTENT=${SKIPINTERMITTENT:-true}
+STATIC=${STATIC:-false}
 
 # ensure Jython can cache JAR classes
 PYTHON_CACHEDIR="${HOME}/jython/cache"
 mkdir -p ${PYTHON_CACHEDIR}
 
 export MAVEN_OPTS=-Xmx1536m
 
+# ensure that working directory is really clean
+mvn antrun:run -Danttarget=realclean
```

Comment:
If this has any effect on the build, that means we are retaining something removed by `realclean` in GitHub, and should remove that from the repo and force it to be ignored by git.

Reply:
Not sure what "remove that from the repo and force it to be ignored by git" means here.  See the jmri-developers discussion: at least .run.sh and the .classes files _where_ being retained,  because the 4th job was running, but it doesn't have a build step.

Reply:
If `.run.sh` and the .classes files were being retained, I'd find them in https://github.com/JMRI/JMRI. `.run.sh` *is* created by an ant process on the fly in `runtest.csh`. If the results of a Java compilation are being retained across Travis CI builds, then we've screwed up our CI build process and shouldn't be trusting any of it anywhere, since that means that builds are not occurring against current code.

Reply:
I'm not following your logic. I'm pretty sure that neither .run.sh nor the .classes files are in Git.

Reply:
Here's an example of the 4th branch successfully running tests that it hasn't compiled:
https://travis-ci.org/JMRI/JMRI/jobs/614207878?utm_medium=notification&utm_source=github_status

I completely agree in advance that I don't understand how Maven does caching.  But I don't think that's the issue here, as in the log linked just above, Maven wasn't even invoked.

Reply:
You are running `ant realclean` on a brand new clean JMRI checkout on a newly built virtual machine that will be destroyed at the end of the job to to prevent it from being reused. If `ant realclean` has *any effect*, then the JMRI git repo has something in it that should not be there.

Reply:
In your example, mvn is run to verify the code at line 353 of the logs. Verification in maven includes compiling, testing, and packaging the code in a JAR, except that at line 353, the testing step is explicitly skipped by other command line arguments.

Reply:
(As you know, I think the hidden-process nature of maven is a problem, not a feature)  Thanks, that explains the .classes files. Can you see where the .run.sh file is created? And why it wasn't created for https://travis-ci.org/JMRI/JMRI/jobs/616148232?utm_medium=notification&utm_source=github_status ?